pkg/cmd/task/create_test.go

```diff
@@ -0,0 +1,183 @@
+			command:     []string{"create", "-f", "../../../test/resources/task.txt", "-n", "ns"},
+			input:       seeds[1],
+			inputStream: nil,
+			wantError:   true,
+			want:        "does not support such extension for ../../../test/resources/task.txt",
+		},
+		{
+			name:        "Mismatched resource file",
+			command:     []string{"create", "-f", "../../../test/resources/taskrun.yaml", "-n", "ns"},
+			input:       seeds[1],
+			inputStream: nil,
+			wantError:   true,
+			want:        "provided TaskRun instead of Task kind",
+		},
+		{
+			name:        "Existing task",
+			command:     []string{"create", "-f", "../../../test/resources/task.yaml", "-n", "ns"},
+			input:       seeds[3],
+			inputStream: nil,
+			wantError:   true,
+			want:        "task \"build-docker-image-from-git-source\" has already exists in namespace ns",
+		},
+	}
+
+	for _, tp := range testParams {
+		t.Run(tp.name, func(t *testing.T) {
+			p := &test.Params{Tekton: tp.input.Pipeline}
+			task := Command(p)
+
+			if tp.inputStream != nil {
+				task.SetIn(tp.inputStream)
+			}
+
+			out, err := test.ExecuteCommand(task, tp.command...)
+			if tp.wantError {
+				if err == nil {
+					t.Errorf("Error expected here")
+				} else {
+					test.AssertOutput(t, tp.want, err.Error())
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Unexpected Error")
+				}
+				test.AssertOutput(t, tp.want, out)
+			}
+		})
+	}
+}
+
+func TestLoadRemoteFile(t *testing.T) {
+	h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		okResponse := `
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: build-docker-image-from-git-source
+spec:
+  inputs:
+    resources:
+      - name: docker-source
+        type: git
+    params:
+      - name: pathToDockerFile
+        type: string
+        description: The path to the dockerfile to build
+        default: /workspace/docker-source/Dockerfile
+      - name: pathToContext
+        type: string
+        description:
+          The build context used by Kaniko
+          (https://github.com/GoogleContainerTools/kaniko#kaniko-build-contexts)
+        default: /workspace/docker-source
+  outputs:
+    resources:
+      - name: builtImage
+        type: image
+  steps:
+    - name: build-and-push
+      image: gcr.io/kaniko-project/executor:v0.9.0
+      # specifying DOCKER_CONFIG is required to allow kaniko to detect docker credential
+      env:
+        - name: "DOCKER_CONFIG"
+          value: "/builder/home/.docker/"
+      command:
+        - /kaniko/executor
+      args:
+        - --dockerfile=$(inputs.params.pathToDockerFile)
+        - --destination=$(outputs.resources.builtImage.url)
+        - --context=$(inputs.params.pathToContext)`
+		_, _ = w.Write([]byte(okResponse))
+	})
+	httpClient, teardown := testingHTTPClient(h)
+	defer teardown()
+
+	cli := NewClient(time.Duration(0))
+	cli.httpClient = httpClient
+	remoteContent, _ := loadFileContent("https://gist.githubusercontent.com/tekton/test.yaml", cli)
```

Comment:
correct me if I am wrong, but this url could be anything (because of the testing client). If that's true, we may want to make the url explicitely something that shouldn't exists (it will be clearer for readers – otherwise first thing that pop in my mind is "let me look at this file on gist", and the 2nd is "we shouldn't do external network call in the unit tests").

Reply:
Make sense.